### PR TITLE
Allow setting column to query

### LINKED ISSUE #18: Filter ignores `useColumn`
Hi,

I have a complex query on a table, on which I would like to use `DateRangeFilter`. Due to the complexity, I have to specify the table and column to be used on the filter. This is my code:

```php
DateRangeFilter::make('created_at')
    ->useColumn('articulos.created_at')
    ->label('Fecha de alta'),
```

There, `useColumn` is ignored and it generates a query like:

```sql
select count(*) as aggregate
from `articulos`
where (
    `articulos`.`deleted_at` is null
    and `created_at` between '2023-05-01 00:00:00' and '2023-05-31 23:59:59'
)
```

instead of

```sql
select count(*) as aggregate 
from `articulos` 
where (
    `articulos`.`deleted_at` is null
    and `articulos`. `created_at` between '2023-05-01 00:00:00' and '2023-05-31 23:59:59'
)
```

It would also be good to allow the developer to modify the query, like the included filters on Filament.

Thanks!

### PR DESCRIPTION
**Pull request by underdpt**
This fixes #18

### CODE REVIEW

File: src/Filters/DateRangeFilter.php [hunk 1]
```diff
@@ -157,13 +155,15 @@ public function useColumn(string $column): self
         return $this;
     }
 
-    public function dateRangeQuery(Builder $query, array $data = []): Builder
```
Comment (malzariey) [inline src/Filters/DateRangeFilter.php:160]: Why was this changed?

File: src/Filters/DateRangeFilter.php [hunk 2]
```diff
@@ -144,8 +144,6 @@ protected function setUp(): void
     {
         parent::setUp();
 
-        $this->useColumn($this->getName());
-        $this->query(fn($query, $data) => $this->dateRangeQuery($query, $data));
```
Comment (malzariey) [inline src/Filters/DateRangeFilter.php:148]: Why was this changed?

**Review by malzariey — COMMENTED** [review]
Please check my comments

File: src/Filters/DateRangeFilter.php (hunk 1 above)
Comment (underdpt) [reply src/Filters/DateRangeFilter.php:160]: To overload `apply` function from trait `InteractsWithTableQuery`, this follows how Filaments plugins work.

If the developer needs to further modify the query, it would need to mimic some of the parent `apply` code (mainly checking `$this->hasQueryModificationCallback()`.

File: src/Filters/DateRangeFilter.php (hunk 2 above)
Comment (underdpt) [reply src/Filters/DateRangeFilter.php:148]: Mmmmm, `useColumn` was overwriting the developer`s call to `useColumn` when configuring the filter. But I see there's an issue if the developer doesn't call `useColumn`, so I have to rethink this.

Let me check if there's another way and i'll update this PR with a better solution.

Comment (underdpt) [reply src/Filters/DateRangeFilter.php:148]: I found the issue causing the developer's custom `column` to be overwritten. See commits below.

Comment (malzariey) [reply src/Filters/DateRangeFilter.php:148]: Let me do a run test later today, then merge. 

Thanks for your efforts!!
